### PR TITLE
chore: enable eslint to unit test v7

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

Introduce test suite lint with eslint and [eslint-plugin-mocha](https://github.com/lo1tuma/eslint-plugin-mocha)

## Notable Changes
only the following items should be reviewed. Other changes are either automatic or verified by eslint itself.
- `configuration` package
- `BasicElement.test.js`
- `ControlElement.test.js`
- `TapEvent.test.js`
- `elf.test.js`
- `canvas.test.js`
- `chart.test.js`
- `interactive-chart.test.js`
- `number-field.test.js`
- `slider.test.js`
- `.eslintrc`
- `.eslintignore`
- `overlay.test.js`
- `overlay.position.test.js`
- `overlay-position-target.test.js`
- `i18n.test.js`
- `test-helpers.test.js`
- `pagination.test.js`
- `tree-select-filter.test.js`
- `color-picker.test.js`
- `label.test.js`
- `multi-input.test.js`
- `focus-manager.test.js`
- `button-bar.test.js`
- `card.test.js`
- `color-palettes.test.js`
- `tree-select.test.js`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/configurations/typescript.eslint-config.js [hunk 1]
```diff
@@ -6,7 +6,6 @@ module.exports = {
       parser: '@typescript-eslint/parser',
       plugins: ['@typescript-eslint', 'eslint-plugin-import'],
       extends: [
-        './eslint-config.js',
```
Comment (wsuwt) [inline packages/configurations/typescript.eslint-config.js:9]: I wonder if this would change how our eslint is setup today ?

Comment (wattachai-lseg) [reply packages/configurations/typescript.eslint-config.js:9]: it doesn't as `extends` on line 2 already applies to both overrides.

**Comment by wsuwt** [comment]
Looks good to me.

**Review by wsuwt — DISMISSED** [review]

File: packages/elements/src/appstate-bar/__test__/appstate-bar.test.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ describe('appstate-bar/AppstateBar', () => {
     );
   });
 
-  it('Should changed heading background to default color when invalid state is set', () => {
+  it('Should changed heading background to default color when invalid state is set', function() {
```
Comment (bualoy-napat) [inline packages/elements/src/appstate-bar/__test__/appstate-bar.test.js]: Should we add `elementUpdated(el)` after line 70 ?

File: packages/elements/src/autosuggest/__test__/autosuggest-functional.test.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ import {
   tapAction
 } from './helpers/helpers.js';
 
-describe('autosuggest/Functional', () => {
-  describe('Functionality', () => {
+describe('autosuggest/Functional', function () {
+  describe('Functionality', function () {
```
Comment (bualoy-napat) [inline packages/elements/src/autosuggest/__test__/autosuggest-functional.test.js:22]: You forget to remove arrow function in line 81

Comment (wattachai-lseg) [reply packages/elements/src/autosuggest/__test__/autosuggest-functional.test.js:22]: somehow eslint couldn't detect this! Fixed.

File: packages/elements/src/appstate-bar/__test__/appstate-bar.test.js (hunk 2 above)
Comment (wattachai-lseg) [reply packages/elements/src/appstate-bar/__test__/appstate-bar.test.js]: good catch! Fixed.

**Comment by wattachai-lseg** [comment]
> You forget to remove arrow function in line 81

**Comment by bualoy-napat** [comment]
LGTM :)

**Review by bualoy-napat — APPROVED** [review]

**Pull request merged**